### ISSUE TITLE
Add a `queryEffectiveChildren` equivalent?

### ISSUE THREAD

**Issue by arthurevans** [Polymer/polymer#4953]
While `FlattenedNodesObserver.getFlattenedNodes` basically provides the same API as `getEffectiveChildren` in 1.x, there's no `queryEffectiveChildren` equivalent. This should be simple-ish with `matches`, but that's not provided as an instance property in 2.x, and it's not polyfilled by shadyDOM/shadyCSS.

Both shady DOM and shady CSS appear to include their own little mini-fill for `matches`, but neither of them expose it as far as I can tell. It is exposed on `Polymer.dom`. So to mimic the old `queryEffectiveChildren` I need to write this:

```
<link rel='import' href='https://polygit.org/components/polymer/lib/utils/flattened-nodes-observer.html'>
<!-- legacy? ick! -->
<link rel='import' href='https://polygit.org/components/polymer/lib/legacy/polymer.dom.html'>

... 
  myQueryFlattenedChildren(selector) {
        var nodes = Polymer.FlattenedNodesObserver.getFlattenedNodes(this);
        return filtered = nodes.filter((node) => node.nodeType === Node.ELEMENT_NODE && Polymer.dom.matchesSelector(node, 'div'));
  }
```

And if I'm importing `Polymer.dom`, I might as well use `Polymer.dom(this).queryDistributedElements(selector)`, which does the same thing (although it should really be called `queryEffectiveChildren`)

In short: I end up writing boilerplate or calling legacy methods... Should we add this to `FlattenedNodesObserver`, or else add `matchesSelector` to something non-legacy? It just seems too darned useful to be buried.

**Comment by arthurevans** [comment]
See also: slack thread: https://polymer.slack.com/archives/C03PF4L4L/p1511099372000036

**Comment by TimvdLippe** [comment]
I would say this is a valid use-case as I have encountered this need some time as well. @kevinpschaaf @sorvell @azakus WDYT?

**Comment by phidias51** [comment]
I recently encountered this. It would be nice if when you deprecate something there was a note to indicate what the replacement code is. Given that the replacement is more verbose, and relies on a legacy API, I would definitely vote for a replacement, or at the very least a new implementation that uses the same method signature.
